### PR TITLE
Ss.clickable dashboard Totals : Connected, Offline, Critical, Warning and Total

### PR DESCRIPTION
**Pull request by swetha571**
# Type of change? 

- [x] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
Clickable dashboard totals : 
1. When user clicks on the Total device count it has to take the user to the Device Page with all total devices grid.
2. If user clicks on the Connected total on dashboard it has to take the user to the devices page but only shows the connected devices grid.
3. If user clicks on the offline total on the dashboard then it has to take the user to the devices page but only shows the offline devices grid.
4. When user clicks on the Critical Alarm total on the dashboard it has to take the user to maintanance page shows only critical alarm alarmByRulesGrid and same will be applicable for the warning totals also.
5. When user clicks on 0 total then user takes to that particular page and shows "No results found".

...

**Checklist:**

- [ ] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/components/pages/devices/devices.js [hunk 1]
```diff
@@ -54,6 +54,20 @@ class DevicesPage extends Component {
...
+  filterDevies(devices) {
+    if (!devices || !devices.Items) {
+      return devices;
+    }
+    return {
+      Items: devices.Items.filter((device) => {
+        if (!this.props.params || !this.props.params.status) {
+          return true;
+        }
+        return device.Connected === (this.props.params.status === 'Connected');
```
Comment (ppathan) [inline src/components/pages/devices/devices.js]: Is 'Connected ' going to be case-sensitive? Can you just compare with case-insensitive?

File: src/components/pages/devices/devices.js [hunk 2]
```diff
@@ -86,7 +100,7 @@ class DevicesPage extends Component {
             </div>
             <div onClick={this.refreshData} className="refresh-icon icon-sm" />
           </div>
-          <DevicesGrid {...deviceGridProps} />
+          { (devices || []).length > 0 ? <DevicesGrid {...deviceGridProps}/>  : <div className="no-results"> No results found </div> }
```
Comment (ppathan) [inline src/components/pages/devices/devices.js]: "No results found" in lang file

File: src/components/maintenance/maintenanceWidget.js [hunk 3]
```diff
@@ -81,7 +103,13 @@ class MaintenanceWidget extends Component {
           <div className={`selection-item${alarmSelected}`} onClick={this.selectGrid}>{lang.NOTIFICATIONS}</div>
           <div className={`selection-item${systemSelected}`} onClick={this.selectGrid}>{lang.JOBS}</div>
         </div>
-        <div className={`grid-container${alarmSelected}`}><AlarmsByRuleGrid {...alarmsByRuleGridProps} /></div>
+        <div className={`grid-container${alarmSelected}`}>
+          {
+            (this.state.alarmsByRuleGridData || []).length > 0
+            ? <AlarmsByRuleGrid {...alarmsByRuleGridProps} />
+            : <div className="no-results">No results found</div>
```
[the remark this answers is not in the document]
Comment (swetha571) [reply src/components/maintenance/maintenanceWidget.js]: If the Critical, warning, total, connected and offline totals on dashboard are 0 then when user clicks on the 0's we show the "no results found".

File: src/components/pages/devices/devices.js (hunk 1 above)
Comment (swetha571) [reply src/components/pages/devices/devices.js]: Yes. 
image

File: src/components/pages/devices/devices.js (hunk 2 above)
Comment (swetha571) [reply src/components/pages/devices/devices.js]: Sure.

File: src/components/pages/devices/devices.js (hunk 1 above)
Comment (ppathan) [reply src/components/pages/devices/devices.js]: My point was if you can do something like this this.props.params.status.toLower() === 'connected' Then we don't have to worry about future scenario where case changes

File: src/components/pages/devices/devices.js [hunk 4]
```diff
@@ -54,6 +54,20 @@ class DevicesPage extends Component {
   /** Listen for changes in the dynamic context filters and update accordingly */
   onContextMenuChange = contextBtns => this.setState({ contextBtns });
 
+  filterDevies(devices) {
```
Comment (spryor) [inline src/components/pages/devices/devices.js]: Typo, `filterDevies` to `filterDevices`

File: src/components/pages/devices/devices.js [hunk 5]
```diff
@@ -54,6 +54,20 @@ class DevicesPage extends Component {
...
+    if (!devices || !devices.Items) {
```
Comment (spryor) [inline src/components/pages/devices/devices.js]: Little enhancement, double check the logic

```
filterDevices(devices) {
	if (!devices || !devices.Items) return devices;
	const status = (this.props.params || {}).status || '';
  	const isConnected = status.toLowerCase() === 'connected'
	return {
	  Items: devices.Items.filter((device) => {
	    if (!status) return true;
	    return device.Connected === isConnected;
	  })
	};
}
```

File: src/components/deviceMap/regionDetails.js [hunk 6]
```diff
@@ -8,13 +9,29 @@ import WarningSvg from '../../assets/icons/Warning.svg';
...
+  constructor() {
+    super();
+    this.criticalAlarmsLink = this.gotoLink.bind(null, '/maintenanceBySeverity/critical');
+    this.warningAlarmsLink = this.gotoLink.bind(null, '/maintenanceBySeverity/warning');
+    this.offlineDeviceLink = this.gotoLink.bind(null, '/devices/NotConnected');
+    this.totalDeviceLink = this.gotoLink.bind(null, '/devices');
+    this.onlineDeviceLink = this.gotoLink.bind(null, '/devices/Connected');
+  }
+
+  gotoLink(linkPath) {
```
[the remark this answers is not in the document]
Comment (swetha571) [reply src/components/deviceMap/regionDetails.js]: No I guess. I still need bind.

**Pull request merged**